File: android/src/main/kotlin/com/kurban/xue_hua_video_snapshot/XueHuaVideoSnapshotPlugin.kt
package com.kurban.xue_hua_video_snapshot

import io.flutter.embedding.engine.plugins.FlutterPlugin

/// Registers the Pigeon [VideoDecoderHostApi] for Android.
class XueHuaVideoSnapshotPlugin : FlutterPlugin {

    private var decoderApi: AndroidVideoDecoderHostApi? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        if (decoderApi == null) {
            decoderApi = AndroidVideoDecoderHostApi(binding.applicationContext)
        }
        VideoDecoderHostApi.setUp(binding.binaryMessenger, decoderApi)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        VideoDecoderHostApi.setUp(binding.binaryMessenger, null)
        decoderApi = null
    }
}

File: ios/xue_hua_video_snapshot/Sources/xue_hua_video_snapshot/XueHuaVideoSnapshotPlugin.swift
import Flutter
import Foundation

/// iOS plugin entry — registers Pigeon [VideoDecoderHostApi].
public class XueHuaVideoSnapshotPlugin: NSObject, FlutterPlugin {
    private let decoderApi = AppleVideoDecoderHostApi()

    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = XueHuaVideoSnapshotPlugin()
        VideoDecoderHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance.decoderApi)
        registrar.addMethodCallDelegate(instance, channel: FlutterMethodChannel(
            name: "xue_hua_video_snapshot",
            binaryMessenger: registrar.messenger()
        ))
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        result(FlutterMethodNotImplemented)
    }
}
